File: Unit01/thermometer.java
/**
 * This file is a thermometer class that can convert temperature from one type of measurement to another.
 * @author Alex Tedesco
 */

public class thermometer {
    /**
    * This class creates a thermometer with a specific temperature.
    */
    
    private double temperature;
    /**
    * This holds the thermometer's temperature.
    */

    thermometer (double temperature) {
        this.temperature = temperature;
    }
    
    public double getTemperature() {
    /**
    * This returns the current temperature of the thermometer.
    */
        return this.temperature;
    }

    public double setTemperature(double temperature) {
    /**
    * This sets a new temperature for the thermometer.
    */
        return this.temperature;
    }

    void change_temp (String units) {
    /**
    * This converts the current temperature of a thermometer from one type of unit to another.
    */
        if (units == "F") {
            this.temperature = (this.temperature * 9/5) + 32;
        } else if (units == "C") {
            this.temperature = this.temperature;
        } else if (units == "K") {
            this.temperature = (this.temperature + 273.15);
        }
        
            
    }

    public static void main(String[] args) {
    /**
    * The main function with all the executed code.
    */
        thermometer th1 = new thermometer (100);
    
        System.out.println(th1.getTemperature() + " degrees");

        th1.change_temp("C");

        System.out.println(th1.getTemperature() + " degrees");

        th1.change_temp("F");

        System.out.println(th1.getTemperature() + " degrees");

            
            }
        
        }
        

File: Unit01/thermometer.py
"""
This file is a thermometer class that can convert temperature from one type of measurement to another.
author: Alex Tedesco
"""

class Thermometer:
    """
    This class creates a thermometer with a specific temperature.
    """
    __slots__ = ["__temperature"]
    def __init__(self, temperature):
        self.__temperature = temperature

    def get_temperature (self):
        '''Returns the current temperature of a created thermometer.'''
        return self.__temperature

    def set_temperature (self, temperature):
        '''Sets the temperature of a created thermometer.'''
        self.__temperature = temperature

    def convert_temp (self, units):
        '''Method that converts the default celsius temperature to the specified temperature units (C, F, K).'''
        if units == "F":
            self.__temperature = (self.__temperature * 9/5) + 32
        elif units == "C":
            self.__temperature = self.__temperature
        elif units == "K":
            self.__temperature = (self.__temperature + 273.15)

    def __str__ (self):
        celsius = self.__temperature
        self.convert_temp ("F")
        fahrenheit = self.__temperature
        self.set_temperature (celsius)
        self.convert_temp ("K")
        kelvin = self.__temperature
        self.convert_temp ("C")
        self.set_temperature (celsius)
        return "Thermometer: {}C {}F {}K".format(str (celsius), str (fahrenheit), str (kelvin))

    def __eq__(self, other):
        if type (self) == type (other):
            return self.__temperature == other.__temperature
        return False

def main ():
    """
    The only executed code in the moduel
    """
    th1 = Thermometer (100)
    # print (th1.get_temperature (), "degrees")

    # th1.convert_temp ("F")
    # print (th1.get_temperature (), "degrees")

    # print (th1)
    th2 = Thermometer (100)
    th3 = Thermometer (101)
    print (th1 == th2) # True
    print (th1 == th3) # False


'''
Used to make sure that, if this program is imported into another module, the
main method will not be executed.
'''
if __name__ == "__main__":
    main ()




